packages/chip-input/src/ux-choice-container-attribute.ts

```diff
@@ -0,0 +1,105 @@
+import { UxChoiceAttribute } from './ux-choice-attribute';
+import { customAttribute, bindingMode, bindable, inject } from 'aurelia-framework';
+
+@customAttribute('ux-choice-container', bindingMode.twoWay)
+@inject(Element)
+export class UxChoiceContainerAttribute {
+
+  @bindable({defaultBindingMode: bindingMode.twoWay, primaryProperty: true}) public value: any | any[];
+  @bindable public multiple: boolean | 'auto' = 'auto';
+
+  public isMultiple: boolean = false;
+  private choices: Array<UxChoiceAttribute> = [];
+  private processClick: (event: Event) => void;
+
+  constructor(private element: Element) {
+  }
+
+  public bind() {
+    this.multipleChanged();
+    this.valueChanged(this.value);
+  }
+
+  public multipleChanged() {
+    if (this.multiple === 'auto') {
+      this.isMultiple = Array.isArray(this.value) ? true : false;
+    } else if (typeof this.multiple === 'boolean') {
+      this.isMultiple = this.multiple;
+    } else {
+      this.isMultiple = this.multiple === 'multiple' || this.multiple === 'true';
+    }
+    this.element.classList.toggle('ux-choice-container--multiple', this.isMultiple);
+  }
+
+  public attached() {
+    this.element.classList.add('ux-choice-container');
+    this.processClick = (event: Event) => {
```

Comment:
we can avoid having to create an arrow function here by using `handleEvent`. It would look like this:
```ts
public attached() {
  this.element.addEventListener('click', this);
}

public detached() {
  this.element.removeEventListener('click', this);
}

public handleEvent(e: Event) {
  if (event.target instanceof Element) {
    const choiceElement: any = event.target.closest('.ux-choice');
    if (
      choiceElement !== null &&
      choiceElement.au !== undefined &&
      choiceElement.au['ux-choice'] !== undefined &&
      choiceElement.au['ux-choice'].viewModel instanceof UxChoiceAttribute) {
      const choice: UxChoiceAttribute = choiceElement.au['ux-choice'].viewModel;
      this.toggleValue(choice.value);
    }
  }
}
```

Reply:
cool trick, didn't know we can pass an object with `handleEvent` method. This is very usefull !

Reply:
for UI lib, this helps avoid a good amount of unnecessary memory